transformations/discourse_marker_substitution/transformation.py

```diff
@@ -0,0 +1,142 @@
+        "accordingly",
+        "hence",
+    ],
+    "Expansion.Alternative.Chosen alternative": ["instead"],
+    "Expansion.Alternative.Conjunctive": ["or"],
+    "Expansion.Conjunction": [
+        "and",
+        "in fact",
+        "while",
+        "also",
+        "furthermore",
+        "in addition",
+        "meanwhile",
+        "moreover",
+        "similarly",
+        "further",
+        "in turn",
+        "likewise",
+        "besides",
+        "additionally",
+    ],
+    "Expansion.Instantiation": ["for example", "for instance"],
+    "Expansion.Restatement.Equivalence": ["in other words"],
+    "Expansion.Restatement.Generalization": ["in sum"],
+    "Expansion.Restatement.Specification": ["specifically", "in particular"],
+    "Temporal.Asynchronous.Precedence": ["then", "subsequently", "finally", "later"],
+    "Temporal.Asynchronous.Succession": ["previously", "earlier", "after"],
+    "Temporal.Synchrony": ["at the time", "when", "at the same time"],
+}
+
+MARKER_TO_CLASS = {
+    "accordingly,": "Contingency.Cause.Result",
+    "additionally,": "Expansion.Conjunction",
+    "also,": "Expansion.Conjunction",
+    "and,": "Expansion.Conjunction",
+    "as a result": "Contingency.Cause.Result",
+    "at the same time": "Temporal.Synchrony",
+    "at the time": "Temporal.Synchrony",
+    "because": "Contingency.Cause.Reason",
+    "besides": "Expansion.Conjunction",
+    "but,": "Comparison.Contrast",
+    "consequently": "Contingency.Cause.Result",
+    "earlier": "Temporal.Asynchronous.Succession",
+    "finally": "Temporal.Asynchronous.Precedence",
+    "for example": "Expansion.Instantiation",
+    "for instance": "Expansion.Instantiation",
+    "further": "Expansion.Conjunction",
+    "furthermore": "Expansion.Conjunction",
+    "hence": "Contingency.Cause.Result",
+    "however,": "Comparison.Contrast",
+    "in addition,": "Expansion.Conjunction",
+    "in fact,": "Expansion.Conjunction",
+    "in particular": "Expansion.Restatement.Specification",
+    "in turn,": "Expansion.Conjunction",
+    "inasmuch as": "Contingency.Cause.Reason",
+    "later": "Temporal.Asynchronous.Precedence",
+    "likewise": "Expansion.Conjunction",
+    "meanwhile": "Expansion.Conjunction",
+    "moreover": "Expansion.Conjunction",
+    "on the contrary": "Comparison.Contrast",
+    "previously": "Temporal.Asynchronous.Succession",
+    "similarly": "Expansion.Conjunction",
+    "since": "Contingency.Cause.Reason",
+    "so,": "Contingency.Cause.Result",
+    "specifically": "Expansion.Restatement.Specification",
+    "subsequently": "Temporal.Asynchronous.Precedence",
+    "then,": "Temporal.Asynchronous.Precedence",
+    "therefore": "Contingency.Cause.Result",
+    "thus": "Contingency.Cause.Result",
+}
+
+
+def discourse_marker_substitution(text, seed=0, max_output=1):
+    random.seed(seed)
+    perturbed_texts = []
+    for _ in range(max_output):
+        present_markers = [m for m in MARKER_TO_CLASS if m in text.lower()]
```

Comment:
Smaller markers like "so", "but" and "and" could appear in larger words like "torso", "homeland", "thousand" and would break your logic. Might be better to extract the words using regex/split and then match.